arangod/Aql/OptimizerRules.cpp

```diff
@@ -4967,17 +4968,17 @@ void arangodb::aql::restrictToSingleShardRule(
         auto collection =
             ExecutionNode::castTo<ModificationNode const*>(current)
                 ->collection();
-        std::string shardId =
+        auto shardId =
             ::getSingleShardId(plan.get(), current, collection);
 
-        if (!shardId.empty()) {
+        if (shardId.has_value() && shardId.value().isValid()) {
```

Comment:
This looks a bit strange. If `getSingleShardId` returns a ShardID value, shouldn't it always return a valid ShardID value? If not, we should maybe change that function accordingly.

Reply:
I agree it either returns nullopt or a valid shard.
Moved to Assertion